### PR TITLE
add tests on rpt destination filtering

### PR DESCRIPTION
**Pull request by uwiger**
Ref https://www.pivotaltracker.com/story/show/154133896

Added unit and system test cases for destination filtering, and indeed found a couple of bugs.

### CODE REVIEW

File: apps/aecore/src/aec_metrics_rpt_dest.erl [hunk 1]
```diff
@@ -170,11 +170,17 @@ make_destinations(Found, Actions0, Tab) ->
     Dests = mk_dests_(Actions0),
     lists:foreach(
       fun({Metrics, DPs}) ->
-              [ets:insert(Tab, #metric{key = {N,D}, destinations = Dests})
+              [ets_insert_new(Tab, #metric{key = {N,D}, destinations = Dests})
                || {N, _, _} <- Metrics,
                   D <- DPs]
       end, Found).
 
+ets_insert(Tab, Obj) ->
```
Comment (happi) [inline apps/aecore/src/aec_metrics_rpt_dest.erl]: Why is this function introduced?
Perhaps it should have a less specific name to allow for future rewrite to use another backend?

Comment (uwiger) [reply apps/aecore/src/aec_metrics_rpt_dest.erl]: It was introduced to simplify debugging via tracing. I usually add a comment to that effect.

**Comment by uwiger** [comment]
I've realized that while the filter (now) seems to work, it doesn't pick up metrics created after the initial population of the filter table (again with the complex start order ...). The system metrics observed by Dincho seem to have been created afterwards.

I'm working on a change in the code to apply the filter rules to metrics if they can't be found in the filter table; the result will then be cached in the table for future reference.

File: apps/aecore/src/aec_metrics_lib.erl [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
 
 find_entries_cmd(Arg) ->
-    find_entries_cmd(Arg, enabled).
+    find_entries_(Arg, enabled, fun exometer:select/1).
 
 find_entries_cmd(Arg, Status) ->
+    find_entries_(Arg, Status, fun exometer:select/1).
+
+filter_entries(Entries, Arg) ->
+    Entries = [{N,D,S} || {N,D,S} <- Entries,
```
Comment (lucafavatella) [inline apps/aecore/src/aec_metrics_lib.erl]: This line silently discards items in `Entries` that do not match `{_,_,_}`, making it fragile to future code changes. May I suggest replacing this with code that hardcodes the expectation on each item of `Entries`.

Comment (uwiger) [reply apps/aecore/src/aec_metrics_lib.erl]: This gave me reason to double-check the implementation, and there are actually a few more things I'd like to correct. Will push a new version a bit later.

**Comment by uwiger** [comment]
Rebased and squashed.

**Pull request merged**